extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java

```diff
@@ -2292,50 +2298,135 @@ private ResultHandle addQueryParam(BytecodeCreator methodCreator,
             ResultHandle client,
             ResultHandle genericType,
             ResultHandle paramAnnotations, int paramIndex) {
-        ResultHandle paramArray;
-        String componentType = null;
 
         AssignableResultHandle result = methodCreator.createVariable(WebTarget.class);
-        BranchResult isValueNull = methodCreator.ifNull(target);
+        BranchResult isValueNull = methodCreator.ifNull(webTarget);
         BytecodeCreator notNullValue = isValueNull.falseBranch();
-        if (type.kind() == Type.Kind.ARRAY) {
-            componentType = type.asArrayType().component().name().toString();
-            paramArray = notNullValue.checkCast(queryParamHandle, Object[].class);
-        } else if (isCollection(type, index)) {
-            if (type.kind() == PARAMETERIZED_TYPE) {
-                Type paramType = type.asParameterizedType().arguments().get(0);
-                if (paramType.kind() == CLASS) {
-                    componentType = paramType.name().toString();
-                }
-            }
-            if (componentType == null) {
-                componentType = DotNames.OBJECT.toString();
+        if (isMap(type, index)) {
+            var resolvesTypes = resolveMapTypes(type, index, jandexMethod);
```

Comment:
just a note, usage of Map.Entry here feels weird to me

Reply:
I personally use it all the time because Java doesn't have something like `Tuple2`. If you want, I can replace it with a custom class

Reply:
No, no need to replace it. Maybe some day we could add some simplistic smallrye project, or something inside Quarkus, that would have tuples.